### PR TITLE
 [flink-consumer-source] fix up streaming to be consistent with batch

### PR DESCRIPTION
**Pull request by ambition119**
fix up streaming to be consistent with batch
Supplementary streaming README.me

### CODE REVIEW

File: examples/flink-consumer-source/pom.xml [hunk 1]
```diff
@@ -66,6 +66,18 @@
       <version>${flink.version}</version>
     </dependency>
 
+    <dependency>
```
Comment (erenavsarogullari) [inline examples/flink-consumer-source/pom.xml]: `flink-streaming-scala_${scala.binary.version}` looks redundant (defined as 2 times)

File: examples/flink-consumer-source/pom.xml [hunk 2]
```diff
@@ -66,6 +66,18 @@
...
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-streaming-scala_${scala.binary.version}</artifactId>
+      <version>${flink.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.flink</groupId>
+      <artifactId>flink-table_${scala.binary.version}</artifactId>
```
Comment (erenavsarogullari) [inline examples/flink-consumer-source/pom.xml]: flink-table_${scala.binary.version} looks redundant as well (defined as 2 times)

File: examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/README.md [hunk 3]
```diff
@@ -95,3 +95,186 @@ $ bin/pulsar-client consume -n 0 -s test test_dest
...
+
+
+### PulsarConsumerSourceWordCountToAvroTableSink
+
+This Flink streaming job is consuming from a Pulsar topic and counting the wordcount in a streaming fashion. The job can write the word count results
+to csv file or another Pulsar topic for json format.
+
+The steps to run the example:
+
+1. Start Pulsar Standalone.
```
Comment (erenavsarogullari) [inline examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/README.md]: `PulsarConsumerSourceWordCount` already explains `Step 1, 2 and 3` so we can define as `Step 1, 2 and 3 are same as above.` in 
```
PulsarConsumerSourceWordCountToAvroTableSink
PulsarConsumerSourceWordCountToJsonTableSink
```
Ref: https://github.com/apache/pulsar/blob/master/examples/flink-consumer-source/src/main/java/org/apache/flink/batch/connectors/pulsar/example/README.md

**Comment by erenavsarogullari** [comment]
@ambition119 Thanks for this PR. Overall LGTM. Just a few minor point.

**Comment by ambition119** [comment]
> @ambition119 Thanks for this PR. Overall LGTM. Just a few minor point.

Thank you for your comments.

**Comment by ambition119** [comment]
@erenavsarogullari Could you help run C++ / Python Tests ? tks.

**Comment by erenavsarogullari** [comment]
rerun cpp tests

File: examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToAvroTableSink.java [hunk 4]
```diff
@@ -103,10 +104,15 @@ public static void main(String[] args) throws Exception {
                 );
 
         tableEnvironment.registerDataStream("wc",wc);
-
         Table table = tableEnvironment.sqlQuery("select * from wc");
+        // note: table schema is consistent with avro
+        table.printSchema();
```
Comment (sijie) [inline examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToAvroTableSink.java]: is this for debugging?

File: examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToAvroTableSink.java [hunk 5]
```diff
@@ -103,10 +104,15 @@ public static void main(String[] args) throws Exception {
...
+//        root
```
Comment (sijie) [inline examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToAvroTableSink.java]:
```suggestion
        // root
        //    |-- count: Long
        //    |-- word: GenericType<org.apache.avro.util.Utf8>
```

File: examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToJsonTableSink.java [hunk 6]
```diff
@@ -104,10 +105,14 @@ public static void main(String[] args) throws Exception {
                         new WordWithCount(c1.word, c1.count + c2.count));
 
         tableEnvironment.registerDataStream("wc",wc);
-
         Table table = tableEnvironment.sqlQuery("select * from wc");
+        table.printSchema();
```
Comment (sijie) [inline examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToJsonTableSink.java]: is this for debugging?

File: examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToJsonTableSink.java [hunk 7]
```diff
@@ -104,10 +105,14 @@ public static void main(String[] args) throws Exception {
...
+//        root
```
Comment (sijie) [inline examples/flink-consumer-source/src/main/java/org/apache/flink/streaming/connectors/pulsar/example/PulsarConsumerSourceWordCountToJsonTableSink.java]: can you fix the indent?

**Comment by ambition119** [comment]
@sijie conflict with the master, Please review it carefully

**Comment by ambition119** [comment]
@sijie  Could you help run C++ / Python Tests ? tks.
```java
[ERROR] Tests run: 27, Failures: 1, Errors: 0, Skipped: 19, Time elapsed: 50.22 s <<< FAILURE! - in org.apache.pulsar.client.impl.BrokerClientIntegrationTest
2018-12-31\T\06:43:33.971 [ERROR] setup(org.apache.pulsar.client.impl.BrokerClientIntegrationTest)  Time elapsed: 10.36 s  <<< FAILURE!
org.apache.pulsar.broker.PulsarServerException: java.io.IOException: Failed to bind Pulsar broker on /0.0.0.0:22138
```

**Comment by sijie** [comment]
@ambition119 at any time, you can comment with "run java8 tests" to trigger running CI.

**Pull request merged**